### ISSUE TITLE
Refactor Consensus::JoinDuringCreation

### ISSUE THREAD

**Issue by jayeshbairagi** [OpenST/compute#83]
- Update the params of Consensus::joinDuringCreation
  - joinDuringCreation(bytes32 _metachainId, address _withdrawalAddress).
- Add a require/modifier that the value of CoreLifetime should be Creation.
- Remove validateJoinParams function.
- Remove isCore require.
- Call reputation.registerjoin.
- Call core.joinDuringCreation(msg.sender).
- if the core.joinDuringCreation(msg.sender) returns true, then call ConsensusGateway.declareKernelOpened(core)

**Comment by benjaminbollen** [comment]
when `minValidator` is reached, `CoreLifetime` should move from `creation` to `genesis` (core.joinDuringCreation(msg.sender) returns true)

**Comment by abhayks1** [comment]
Started working on this ticket ⛑

**Comment by abhayks1** [comment]
Moving this ticket to Backlog as couldn't progress on the ticket and had to switch to a priority issue #102 cc: @deepesh-kn

**Comment by abhayks1** [comment]
Started working on this ticket...

**Comment by abhayks1** [comment]
Few of the ticket items are outdated. Updating details here:

- Add a require/modifier that the value of CoreLifetime should be Creation.
  **Done in PR #69** 

- Remove isCore require. 
  **Done in PR #69**

- `Call reputation.registerjoin.`
  Since the Reputation method name is renamed to `stake`,  `reputation.stake` method is called.

- `if the core.joinDuringCreation(msg.sender) returns true`
Core.joinDuringCreation returns validatorCount and minValidatorCount. Updated logic: 
```
if (validatorCount >= minValidatorCount) {
    coreLifetimes[core] = CoreLifetime.genesis;
}
```
- `ConsensusGateway.declareKernelOpened(core)`
I think this should be implemented in issue #89 with the implementation of `ConsensusGateway` contract.

**Question**:
@jayeshbairagi Why do we need to remove validateJoinParams method? 
It's called from `Consensus.joinDuringCreation` and `Consensus.join` and does validation checks.

**Comment by jayeshbairagi** [comment]
- We can have an interface for `ConsensusGateway` and a function `declareKernelOpened` defined for now.

- We can remove `validateJoinParams` method as it can be reduced to just one modifier
```solidity
require(
    _core != address(0),
    "Core address is 0."
);
```
We don’t need 
```solidity
require(
    _metachainId != bytes32(0),
    "Metachain Id id is 0."
);
```
As we are now deriving the `_core` using the `_metachainId`
And we also don’t need 
```solidity
require(
    _withdrawalAddress != address(0),
    "Withdrawal address is 0."
);
```
As this is being checked in `reputation.stake` in the next step

- For the rest, follow as you have mentioned
